src/Raven.Server/Documents/ETL/Providers/AI/GenAi/GenAiBatchPatchCommand.cs

```diff
@@ -48,63 +53,75 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
+                    statsScope.NumberOfContextObjects++;
 
-                    tuple = (document, []);
-                }
+                    if (item.ContextOutput.IsCached)
+                        statsScope.TotalCachedContexts++;
 
-                tuple.Hashes.Add(item.ContextOutput.AiHash);
+                    if (item.UpdateHash == false)
+                        continue;
+                    
+                    ref var tuple = ref CollectionsMarshal.GetValueRefOrAddDefault(hashes, item.DocId, out var exists);
+                    if (exists is false)
+                    {
+                        Document document = GetCurrentDocument(context, item.DocId);
+                        if (document is null)
+                            continue; // document was probably deleted while we talked to the model, skipping this
 
-                if (item.ModelOutput is null)
-                    continue;
+                        tuple = (document, []);
+                    }
 
-                var args = CreatePatchArgs(context, item);
-                try
-                {
-                    var documentInstance = (BlittableObjectInstance)runner.Translate(context, tuple.Doc).AsObject();
-                    using (var scriptResult = runner.Run(context, context, "execute", item.DocId, [documentInstance, args]))
-                    using (var old = tuple.Doc.Data)
+                    tuple.Hashes.Add(item.ContextOutput.AiHash);
+
+                    if (item.ModelOutput is null)
+                        continue;
+                    
+                    statsScope.TotalUpdates++;
+
+                    var args = CreatePatchArgs(context, item);
+                    try
                     {
-                        tuple.Doc.Data = scriptResult.TranslateToObject(context);
+                        var documentInstance = (BlittableObjectInstance)runner.Translate(context, tuple.Doc).AsObject();
+                        using (var scriptResult = runner.Run(context, context, "execute", item.DocId, [documentInstance, args]))
+                        using (var old = tuple.Doc.Data)
+                        {
+                            tuple.Doc.Data = scriptResult.TranslateToObject(context);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        // do not update metadata hash, log error, raise alert
+                        tuple.Hashes.Remove(item.ContextOutput.AiHash);
+                        var msg = $"Failed to apply update script for context in document '{item.DocId}'. " +
+                                  $"Context was: {item.ContextOutput.Context}{Environment.NewLine}" +
+                                  $"Error: {e}";
+
+                        statsScope.UpdateFailures++;
+                        _statistics.RecordPartialLoadError(msg, item.DocId);
+                        _logger.Log(LogLevel.Warn, msg);
```

Comment:
```
if(_logger.IsWarnEnabled)
   _logger.Warn(..)
```

Reply:
please check how it is done in other places and do the same.
(also fix all other relevant places here)